### DISCUSSION TITLE
[WASM] Is it possible to run a uno platform webassembly in Blazor (server side)?

### DISCUSSION THREAD

**Discussion by tobizk04** [unoplatform/uno#16380] [Q&A]
The problem behind it:
We are developing a Uno platform browser app with services behind it and are having
in some difficulties with the memory limitations on mobile devices, especially IOS.

The idea:
In this case, we will run the Uno Platform web assembly on the server.
Blazor server-side rendering.

**Is it possible to embed the Uno Platform into a Razor page on the Blazor server?**
**Has anyone ever tried this?**

**Comment by francoistanguay** [comment]
No, Uno uses .NET for WebAssembly which is client-side only.

FWIW, Uno 5.1 had improvements in terms of memory consumption. If you have a memory-specific issue with a repro, please open a discussion or issue and we could look into it.

Thank you!

**Comment by tobizk04** [comment]
Hi @francoistanguay,
Thanks for your fast answer.

I think that is not a uno platform issue.
The issue is that browser on IOS get not so much memory.

Currently we port our rendering engine from WPF over WinUI3 to Uno and include this in UNO App.
This engine plus the complete app need for some cases more than 1,5 GB memory.
In a browser based on Windows or MAC this is not a problem.
But in case of IOS there is an memory limit.

This memory limits ends in an exception and I have no idea how could we handle this.
We need the memory, otherwise the app could not handle all functionality.
After this exception the app have an undefined status and is not work correctly because for each
additional action we need memory.

When you say we should improve memory usage.
Yes, I know we need to improve this
But at the moment we have decided to accept the additional consumption
in order to make faster progress with the port to uno.

The behaviour in the IOS environment is unpredictable.
Our application is project data-driven.
This means that if the customer makes complex drawings, this leads to higher memory consumption.

**Is there a way to handle this memory limitation?**
**Or extend the memory limitations?**
At the moment, after my research, I still have no idea what exactly the limitation is.
Is the limitation IOS version dependent, also browser dependent or hardware dependent?
Can you understand my problem?

**I would be grateful for any ideas.**

**Reply by jeromelaban** [reply]
There's no explicit way for adjusting memory requirements in Safari at this time. That being said, do you know what consumes memory? You can use https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-profiling.html to troubleshoot further managed memory consumption.
